File: fisica/java/ejercicio3.java
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        String[] nombres = {"Juan", "María", "Luis", "Ana", "Pedro", "Elena"};
        String[][] mitades = dividirLista(nombres);
        
        System.out.println("Primera mitad: " + Arrays.toString(mitades[0]));
        System.out.println("Segunda mitad: " + Arrays.toString(mitades[1]));
    }

    public static String[][] dividirLista(String[] lista) {
        
        int mitad = lista.length / 2;

        
        String[] primeraMitad = Arrays.copyOfRange(lista, 0, mitad);
        String[] segundaMitad = Arrays.copyOfRange(lista, mitad, lista.length);

        return new String[][] {primeraMitad, segundaMitad};
    }
}

File: fisica/js/ejercicio3.js
function dividirLista(lista) {
    // Calcula el índice de la mitad de la lista
    const mitad = Math.floor(lista.length / 2);

    // Obtiene la primera mitad de la lista desde el principio hasta el índice de la mitad
    const primeraMitad = lista.slice(0, mitad);
    // Obtiene la segunda mitad de la lista desde el índice de la mitad hasta el final
    const segundaMitad = lista.slice(mitad);

    // Retorna un arreglo con las dos mitades de la lista
    return [primeraMitad, segundaMitad];
}

// Lista de nombres
const nombres = ["Juan", "María", "Luis", "Ana", "Pedro", "Elena"];
// Llama a la función dividirLista con la lista de nombres como argumento
const [primeraMitad, segundaMitad] = dividirLista(nombres);
// Imprime la primera y segunda mitad de la lista en la consola
console.log("Primera mitad:", primeraMitad);
console.log("Segunda mitad:", segundaMitad);
